File: 01_Language/01_Functions/php/chop.php
<?php

// chop() = rtrim()
$text = "\t\tThese are a few words :) ...  ";
$binary = "\x09Example string\x0A";
$hello  = "Hello World";
var_dump($text, $binary, $hello);

print "\n";

$trimmed = chop($text);
var_dump($trimmed);

$trimmed = chop($text, " \t.");
var_dump($trimmed);

$trimmed = chop($hello, "Hdle");
var_dump($trimmed);

// trim the ASCII control characters at the end of $binary
// (from 0 to 31 inclusive)
$clean = chop($binary, "\x00..\x1F");
var_dump($clean);

File: 01_Language/01_Functions/python/chop.py
# coding: utf-8

text = "\t\tThese are a few words :) ...  "
# binary = "\x09Example string\x0A"
hello = "Hello World"
print(repr(text))
# print(repr(binary))
print(repr(hello))

print(repr(text.rstrip()))
print(repr(text.rstrip(" \t.")))
print(repr(hello.rstrip("Hdle")))

# trim the ASCII control characters at the end of $binary
# (from 0 to 31 inclusive)
# print(repr(binary.rstrip("\x00..\x1F")))
